src/Surfnet/StepupRa/RaBundle/Controller/SecondFactorController.php

```diff
@@ -52,19 +52,17 @@ public function searchAction(Request $request)
         $identity = $this->getCurrentUser();
         $this->get('logger')->notice('Starting search for second factors');
 
-        $institutionFilterOptions = $this
-            ->getInstitutionConfigurationOptionsService()
-            ->getAvailableInstitutionsFor($identity->institution);
-
         $command = new SearchRaSecondFactorsCommand();
         $command->actorInstitution = $identity->institution;
         $command->actorId = $identity->id;
         $command->pageNumber = (int) $request->get('p', 1);
         $command->orderBy = $request->get('orderBy');
         $command->orderDirection = $request->get('orderDirection');
+
+        $secondFactors = $this->getSecondFactorService()->search($command);
         
         // The options that will populate the institution filter choice list.
-        $command->institutionFilterOptions = $institutionFilterOptions;
+        $command->institutionFilterOptions = $secondFactors->getFilterOption('institution');
```

Comment:
Please update the type hint for: https://github.com/OpenConext/Stepup-RA/blob/3d7f0ab4d549971468895f3a26ccc08a89ccfdd7/src/Surfnet/StepupRa/RaBundle/Command/SearchRaSecondFactorsCommand.php#L109 This can now be either array or null.

Also I'm not sure the SearchType will like it if the filter options are `null` . Not sure how the select options would be populated then..

Reply:
.If it's null I would like an exception, because then we use an incompatible Middleware.

Reply:
I ran into this actually, while running a behat test (fga-use-case-c). Maybe there is a bug in the filtering logic for that institution field?

Reply:
I've fixed this by always returning an array for flter options.